File: Project1/Project1.py
"""
A version of Project1.java as a python script.
"""

from random import Random

# since this is python, after all, it'll have a much more "scripty" feel; i.e: prompts.
print 'PyShamBo?'
user_name = raw_input('What\'s your name?\n')
user_hand = raw_input('Now let\'s see if you can beat me.\nWhat\'s your hand?\n')

# same as the r = Math.random() and pcHand assignment block (main:27-37)
r = Random()
x = r.randint(1, 3)

if x==1:
    pc_hand = 'rock'
elif x==2:
    pc_hand = 'scissors'
elif x==3:
    pc_hand = 'paper'

# compare hands
if user_hand == pc_hand:
    winner = 'Tie!'
elif user_hand == 'rock':
    if pc_hand == 'scissors':
        winner = user_name + ' wins!'
    else:
        winner = 'The computer wins!'
elif user_hand == 'scissors':
    if pc_hand == 'rock':
        winner = 'The computer wins!'
    else:
        winner = user_name + ' wins!'
elif user_hand == 'paper':
    if pc_hand == 'rock':
        winner = user_name + ' wins!'
    else:
        winner = 'The computer wins!'
else:
    winner = 'dammit...'

# ...and, output.
print '************************************************************'
print 'User name: %s' % user_name
print 'User hand: %s' % user_hand
print 'Computer hand: %s' % pc_hand
print winner
print '************************************************************'

File: Project1/Project1.java
package Project1;
/**
 *	Project 1: Rock Paper Scissors
 *		This program uses two terminal arguments as input
 *		for a randomized game of Rock, Paper, Scissors.
 *
 *	@author Michael DeWitt
 *	Course	 : CS 177
 *	Section  : 0901
 *	@date 9/14/2009
 */

public class Project1 {
	public static void main (String[] args) {
		/* Instantiates the variables for the user's 
		 * name <userName> and hand <userHand> as well
		 * as the PC's hand <pcHand> and the <winner>
		 */
		String userName,
			   userHand,
			   pcHand,
			   winner;
		
		/* Declares <userName> and <userHand> as 
		 * arguments one and two, respectively
		 */
		userName = args[0];
		userHand = args[1];
		
		/* Uses <Math.random()> and if-else statements
		 * to determine an integer value for <pcHand>
		 */
		double r = Math.random();
		
		if (r < 1/3.0)
			pcHand = "rock";
		else if (r < 2/3.0)
			pcHand = "scissors";
		else
			pcHand = "paper";
		
		/* Compares <pcHand> with <userHand> and declares
		 * a winner via rules of Rock, Paper, Scissors
		 */
		if (userHand.equalsIgnoreCase(pcHand))
			winner = "Tie!";
		else if (userHand.equalsIgnoreCase("rock"))
			if (pcHand.equals("scissors"))
				winner = userName + " wins!";
			else
				winner = "The computer wins!";
		else if (userHand.equalsIgnoreCase("scissors"))
			if (pcHand.equals("rock"))
				winner = "The computer wins!";
			else
				winner = userName + " wins!";
		else if (userHand.equalsIgnoreCase("paper"))
			if (pcHand.equals("rock"))
				winner = userName + " wins!";
			else
				winner = "The computer wins!";
		else
			winner = "Michael can't write Java anymore..."; 	// added to make Eclipse warning about possible null variable go away
		
		// And finally, output:
		System.out.println("Rock Paper Scissors\n"
						 + "************************************************************\n"
						 + "Player name: " + userName + "\n"
						 + "Player hand: " + userHand + "\n"
						 + "Computer generated value: " + r + "\n"
						 + "Computer hand: " + pcHand + "\n"
						 + winner + "\n"
						 + "************************************************************");
	}
}